lib/plugins/auto-load.js

```diff
@@ -3,40 +3,44 @@
 const _ = require('lodash');
 const debug = require('debug')('voxa');
 
+let defaultConfig = { loadByToken: false };
+
 function autoLoad(skill, config) {
-  skill.onRequestStarted((alexaEvent) => {
-    if (!config) {
-      throw Error('Empty config file');
-    }
+  if (!config) {
+    throw Error('Empty config file');
+  }
 
-    if (!config.adapter) {
-      throw Error('Empty adapter');
-    }
+  if (!config.adapter) {
+    throw Error('Empty adapter');
+  }
+
+  if (!_.isFunction(config.adapter.get)) {
+    throw Error('No get method to fetch data from');
+  }
 
-    if (config.loadByToken && !_.get(alexaEvent, 'session.user.accessToken')) {
+  defaultConfig = _.merge(defaultConfig, config);
+
+  skill.onRequestStarted((alexaEvent) => {
+    if (defaultConfig.loadByToken && !_.get(alexaEvent, 'session.user.accessToken')) {
       debug('Missing token on load data by token operation');
       return alexaEvent;
     }
 
     let key = _.get(alexaEvent, 'user.userId');
 
-    if (config.loadByToken) {
+    if (defaultConfig.loadByToken) {
```

Comment:
Remember other programers will have to code the adapter, and the most straightforward the better, that's why I just pass the actual key to the GET method. As a programmer I wouldn't expect a model.user object to then validate that. I would just like to have available the key to fetch with. Thoughts on that approach?